src/main/java/org/opensearch/geospatial/action/upload/geojson/UploadGeoJSONRequestContent.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.geospatial.action.upload.geojson;
+
+import static java.util.Objects.requireNonNull;
+import static org.opensearch.geospatial.GeospatialParser.extractValueAsString;
+
+import java.util.Map;
+
+import org.opensearch.common.ParseField;
+
+public final class UploadGeoJSONRequestContent {
+
+    public static final ParseField FIELD_INDEX = new ParseField("index", new String[0]);
+    public static final ParseField FIELD_GEOSPATIAL = new ParseField("field", new String[0]);
+    public static final ParseField FIELD_DATA = new ParseField("data", new String[0]);
+    private final String indexName;
+    private final String geospatialFieldName;
+    private final Object data;
+
+    private UploadGeoJSONRequestContent(String indexName, String geospatialFieldName, Object data) {
+        this.indexName = requireNonNull(indexName);
```

Comment:
I suggest we call requireNonNull with second param and put some explicit message about what param value shouldn't be null

Reply:
Ack